cpp/ql/src/semmle/code/cpp/ir/implementation/aliased_ssa/Instruction.qll

```diff
@@ -368,6 +369,12 @@ class Instruction extends Construction::TInstruction {
     )
   }
 
+  int getLineRank() {
+    this = rank[result](Instruction instr | instr.getAST().getFile() = getAST().getFile() and instr.getAST().getLocation().getStartLine() = getAST().getLocation().getStartLine() | instr
+     order by instr.getBlock().getDisplayIndex(), instr.getDisplayIndexInBlock()
```

Comment:
The `getDisplayIndexInBlock` predicate is expensive, using both a `count` and a `rank`. Since it's now only used for `order by`, it no longer has to be contiguous or zero-based, so I think you can get rid of both `count` and `rank` by assigning the negative numbers to phi instructions and positive numbers to non-phi instructions. It can also become `private`.

Reply:
I thought you could avoid sorting on `getUniqueId`, but I can see now that it's not so simple because its result is a `string`, not an `int`. Also, this is not performance-sensitive code since it's only for debugging.

Feel free to disregard this suggestion if you think it adds more confusion than clarity (the order of phi nodes got reversed).

Reply:
I do eventually want to refactor the display index code to avoid relying on string comparison, but that's mostly for stable sorting when instructions are added/removed. The phi ordering isn't particularly helpful as-is, so I'm happy to change it.